builtin/providers/aws/resource_aws_rds_cluster.go

```diff
@@ -579,6 +601,27 @@ func resourceAwsRDSClusterUpdate(d *schema.ResourceData, meta interface{}) error
 		}
 	}
 
+	if d.HasChange("iam_roles") {
+		oraw, nraw := d.GetChange("iam_roles")
+		o := oraw.([]interface{})
+		n := nraw.([]interface{})
+		create, remove := rdsClusterDiffIamRoles(o, n)
+
+		for _, role := range create {
```

Comment:
Are you aware of any reason where 2 roles could conflict, such that adding the new ones before removing the old ones, could cause a conflict or failure here?